src/index.ts

```diff
@@ -653,6 +656,146 @@ export function HTPipeDoWork<
   }
 }
 
+// left has sanitizeParams and right has sanitizeParams
+export function HTPipeSanitizeParams<
+  TLeft extends HasSanitizeParams<
+    any,
+    TRight extends HasSanitizeParams<any, any>
+      ? Pick<
+          Parameters<TRight['sanitizeParams']>[0],
+          keyof ReturnType<
+            TLeft extends HasSanitizeParams<any, any>
+              ? TLeft['sanitizeParams']
+              : () => {}
+          >
+        >
+      : any
+  >,
+  TRight extends HasSanitizeParams<any, any>,
+  TContextInLeft extends Parameters<TLeft['sanitizeParams']>[0],
+  TContextInRight extends Parameters<TRight['sanitizeParams']>[0],
+  TContextOutLeft extends ReturnType<TLeft['sanitizeParams']>,
+  TContextOutRight extends ReturnType<TRight['sanitizeParams']>
+>(
+  left: TLeft,
+  right: TRight
+): HasSanitizeParams<
+  TContextInLeft &
+    Omit<
+      TContextInRight,
+      TContextOutLeft extends void ? keyof {} : keyof TContextOutLeft
+    >,
```

Comment:
the return side should be only TContextOutRight
- [ ] fix typescript-side snowballing on all sanitizers

Reply:
actually... i think the input-side is kind of not right too.  I think it should probably be only TContextInLeft ... It should be illegal to have a "right" that needs more than left provides.
- [ ] piper's return value's input side
- [ ] fix the type constraints above accordingly too (it should probably be simpler)

Reply:
done here: https://github.com/captaincaius/hipthrusts/pull/44/files